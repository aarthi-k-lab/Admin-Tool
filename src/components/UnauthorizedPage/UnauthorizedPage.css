.error-icon.error-icon {
  color: red;
  font-size: 10em;
}

.column {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.message {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  font-size: 1.25em;
}
