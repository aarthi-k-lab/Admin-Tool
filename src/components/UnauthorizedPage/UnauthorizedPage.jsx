import React from 'react';
import PropTypes from 'prop-types';
import ErrorIcon from '@material-ui/icons/Error';
import AppCenterDisplay from '../AppCenterDisplay';
import UnauthorizedAppHeader from '../UnauthorizedAppHeader';
import './UnauthorizedPage.css';
import Auth from '../../lib/Auth';

const UnauthorizedPage = ({ location }) => {
  const message = Auth.getErrorMessage(location);
  return (
    <AppCenterDisplay header={<UnauthorizedAppHeader />}>
      <div styleName="column">
        <ErrorIcon styleName="error-icon" />
        <span styleName="message">
          <span>
            {message.text}
          </span>
          <br />
          <span>
            {message.code}
          </span>
        </span>
      </div>
    </AppCenterDisplay>
  );
};

UnauthorizedPage.propTypes = {
  location: PropTypes.shape({
    search: PropTypes.string.isRequired,
  }).isRequired,
};

export default UnauthorizedPage;
