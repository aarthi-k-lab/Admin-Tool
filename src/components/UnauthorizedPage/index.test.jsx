import React from 'react';
import renderer from 'react-test-renderer';
import UnauthorizedPage from '.';

describe('<UnauthorizedPage />', () => {
  test('snapshot test', () => {
    const location = { search: '' };
    const tree = renderer
      .create(<UnauthorizedPage location={location} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});