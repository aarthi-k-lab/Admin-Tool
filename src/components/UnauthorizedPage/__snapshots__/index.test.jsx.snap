// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnauthorizedPage /> snapshot test 1`] = `
<div
  className="components-AppContainer-AppContainer__app"
>
  <header
    className="components-UnauthorizedAppHeader-UnauthorizedAppHeader__header"
  >
    <img
      alt="logo"
      src="/static/img/logo.png"
    />
    <span
      className="components-UnauthorizedAppHeader-UnauthorizedAppHeader__spacer"
    />
  </header>
  <div
    className="components-Body-Body__body-container"
  >
    <section
      className="components-Body-Body__body"
    >
      <div
        className="components-Center-Center__center--expand components-Center-Center__center"
      >
        <div
          className="components-AppCenterDisplay-AppCenterDisplay__app-center-display"
        >
          <div
            className="components-UnauthorizedPage-UnauthorizedPage__column"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-1 components-UnauthorizedPage-UnauthorizedPage__error-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1 15h-2v-2h2v2zm0-4h-2V7h2v6z"
              />
            </svg>
            <span
              className="components-UnauthorizedPage-UnauthorizedPage__message"
            >
              <span>
                You have not been assigned to the app. Kindly contact the support team.
              </span>
              <br />
              <span />
            </span>
          </div>
        </div>
      </div>
    </section>
  </div>
  <footer
    className="components-Footer-Footer__footer"
  >
    Â© Nationstar Mortgage LLC, All rights reserved
  </footer>
</div>
`;
